es6/controls.js

```diff
@@ -7,81 +7,142 @@ import {ExpandableSection} from 'monadical-react-components'
+                        width: `${Math.min((most_future_time - genesis_time) / 150, 100)}%`,
+                    }}/>
+        </div>
         <br/>
-        <Button onClick={setSpeed.bind(this, -100)}>-100x</Button> &nbsp;
-        <Button onClick={setSpeed.bind(this, -10)}>-10x</Button> &nbsp;
-        <Button onClick={setSpeed.bind(this, -1)}>-1x</Button> &nbsp;
-        <Button onClick={setSpeed.bind(this, -0.1)}>-0.1x</Button> &nbsp;
-        <Button onClick={setSpeed.bind(this, -0.01)}>-0.01x</Button> &nbsp;
-        <Button bsStyle="danger" onClick={setSpeed.bind(this, 0)}>⏸</Button> &nbsp;
-        <Button bsStyle="success" onClick={setSpeed.bind(this, 1)}>▶️</Button> &nbsp;
-        <Button onClick={setSpeed.bind(this, 0.01)}>+0.01x</Button> &nbsp;
-        <Button onClick={setSpeed.bind(this, 0.1)}>+0.1x</Button> &nbsp;
-        <Button onClick={setSpeed.bind(this, 1)}>1x</Button> &nbsp;
-        <Button onClick={setSpeed.bind(this, 10)}>+10x</Button> &nbsp;
-        <Button onClick={setSpeed.bind(this, 100)}>+100x</Button>
+        <Button bsStyle={speed === -10 ? 'success' : 'default'}
+                onClick={() => setSpeed(-10)}>-10x</Button> &nbsp;
+        
+        <Button bsStyle={speed === -1 ? 'success' : 'default'}
+                onClick={() => setSpeed(-1)}>-1x</Button> &nbsp;
+        
+        <Button bsStyle={speed === -0.1 ? 'success' : 'default'}
+                onClick={() => setSpeed(-0.1)}>-0.1x</Button> &nbsp;
+        
+        <Button bsStyle={speed === -0.01 ? 'success' : 'default'}
+                onClick={() => setSpeed(-0.01)}>-0.01x</Button> &nbsp;
+        
+        <Button bsStyle={speed === -0.001 ? 'success' : 'default'}
+                onClick={() => setSpeed(-0.001)}>-0.001x</Button> &nbsp;
+
+        {speed === 0 ?
+            <Button bsStyle="success" onClick={() => setSpeed(1)}>▶️</Button>
+          : <Button bsStyle="danger" onClick={() => setSpeed(0)}>⏸</Button>} &nbsp;
+
+        <Button bsStyle={speed === 0.001 ? 'success' : 'default'}
+                onClick={() => setSpeed(0.001)}>+0.001x</Button> &nbsp;
+
+        <Button bsStyle={speed === 0.01 ? 'success' : 'default'}
+                onClick={() => setSpeed(0.01)}>+0.01x</Button> &nbsp;
+
+        <Button bsStyle={speed === 0.1 ? 'success' : 'default'}
+                onClick={() => setSpeed(0.1)}>+0.1x</Button> &nbsp;
+
+        <Button bsStyle={speed === 1 ? 'success' : 'default'}
+                onClick={() => setSpeed(1)}>1x</Button> &nbsp;
+
+        <Button bsStyle={speed === 10 ? 'success' : 'default'}
+                onClick={() => setSpeed(10)}>+10x</Button>
     </ExpandableSection>
 }
 
+export class Ticker {
+    constructor(subscribers, running) {
+        this.subscribers = subscribers || []
+        this.running = running || true
+        this.tick()
+    }
+    
+    subscribe(fn) {
+        this.subscribers.push(fn)
+    }
+
+    tick() {
+        this.subscribers.forEach((fn) => fn())
```

Comment:
```javascript
this.subscribers.map(fn => fn())
```

Reply:
`forEach` is for iterating over things and producing side effects (like here)

`map` is for producing a new list that is a function of the items of the original list (not like here)